### ISSUE TITLE
Rationalize project creation

### ISSUE THREAD

**Issue by snicoll** [spring-projects-experimental/spring-cli#104]
In the current reference guide, the `spring boot new` command is [described as](https://docs.spring.io/spring-cli/reference/creating-new-projects.html):

> The spring boot new command makes it easy to create Spring Boot applications.

Then later on there is the "initializr" command that is [described as](https://docs.spring.io/spring-cli/reference/initializr.html):

>  Shell style project creation mimics how project is created from an UI[...]

I don't really expect those two commands to merge as they're operating with different assumptions. However, I expect the "Spring CLI" to present how to create a new project consistently. Right now the doc has one front and center section for the "boot new" command and what seems to be a generic command doc for the "initializr" one.

Somewhat related and this can be a separate issue if needed, the "initializr" term does not look a great candidate for this IMO. The project code name has a typo "on purpose" but I think it would be confusing for users that are not aware of that (and, frankly, I don't think they should). Using `init` would also be a natural way for users migrating from the Spring (Boot) CLI.

In summary, the CLI offers two ways of creating a new project and no guidance of when to chose one over the other.

**Comment by jvalkeal** [comment]
Ups, had a wrong commit reference, reopen.

**Comment by jvalkeal** [comment]
We've been struggling with "initializr" for reasons you also mentioned and secondly as it's "typoed" it's hard to write. It's definitely going to get renamed/merged with something else. Naming is hard...

**Comment by jvalkeal** [comment]
What I'd really like to see is having `spring init <type>` command.

```
spring init catalog
spring init start
```

We're essentially getting something from a catalog or from an initializer.
